### DISCUSSION TITLE
Bash one liner if statement is not working

### DISCUSSION THREAD

**Discussion by Jojoooo1** [community/community#25399] [Actions]
<p>Hi,</p>
<p>The following statement cause github action to exit:</p>
<pre><code>[["${#TAG_LIST[@]}" -ne 3 ]] &amp;&amp; echo "$RELEASE_VERSION is not a valid semantic version" &amp;&amp; exit 1
</code></pre>
<p>The following statement passes</p>
<pre><code>if [["${#TAG_LIST[@]}" -ne 3 ]]; then
  echo "$RELEASE_VERSION is not a valid semantic version"
  exit 1
fi
</code></pre>
<p>Is there any specific reason for that ?</p>
<p>Best regards,</p>
<p>Jonathan.</p>

**Comment by Yanjingzhu** [comment]
<p>Hi <a class="mention" href="/jojoooo1">@jojoooo1</a> , </p>
<p>When I try the second method, the workflow run also exit. Could you please share your whole workflow yml content here? </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/6/6d74940ce9b1c07ba618c8b0573bbc3c7ffc186a.png" data-download-href="/uploads/short-url/fChOm1HXF6e3dD4nH6sOSykWLRE.png?dl=1" title="bash.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181090757-bbdd87ec-ccc1-4825-bd85-a5cdcee60a77.png" alt="bash.png" data-base62-sha1="fChOm1HXF6e3dD4nH6sOSykWLRE" width="667" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/6/6d74940ce9b1c07ba618c8b0573bbc3c7ffc186a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">bash.png</span><span class="informations">880×659 19.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Is the result of </p>
<pre><code>"${#TAG_LIST[@]}" -ne 3
</code></pre>
<p> changed in your side? </p>

**Comment by Jojoooo1** [comment]
<p>Hi <a class="mention" href="/yanjingzhu">@yanjingzhu</a> </p>
<p>I wrote a bit quickly you will find below the variable details:</p>
<pre><code>RELEASE_VERSION=2.3.1
TAG_LIST=(`echo $RELEASE_VERSION | tr '.' ' '`)

# this will pass
if [["${#TAG_LIST[@]}" -ne 3 ]]; then
  echo "$RELEASE_VERSION is not a valid semantic version"
  exit 1
fi

# this will not pass
[["${#TAG_LIST[@]}" -ne 3 ]] &amp;&amp; echo "$RELEASE_VERSION is not a valid semantic version" &amp;&amp; exit 1
</code></pre>

**Comment by Yanjingzhu** [comment]
<p><a class="mention" href="/jojoooo1">@jojoooo1</a><br>
Thank you for sharing your detail bash scripts here. Based on my test ,</p>
<pre><code class="lang-auto">TAG_LIST=(`echo $RELEASE_VERSION | tr '.' ' '`)
</code></pre>
<p>just returns 2, this cause the result of if condition to false, so exit 1 will not execute when you use if.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/6/68ab30001ffad2f479715718f82b9c2dcd13ae89.png" data-download-href="/uploads/short-url/eVWqjCeluyvAJQ58zrRtsMccB4t.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181090760-bcabd0dd-8e42-4d9f-b83b-90bf892aa864.png" alt="image" data-base62-sha1="eVWqjCeluyvAJQ58zrRtsMccB4t" width="690" height="156" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/6/68ab30001ffad2f479715718f82b9c2dcd13ae89_2_690x156.png, https://aws1.discourse-cdn.com/github/original/2X/6/68ab30001ffad2f479715718f82b9c2dcd13ae89.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/6/68ab30001ffad2f479715718f82b9c2dcd13ae89.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/6/68ab30001ffad2f479715718f82b9c2dcd13ae89_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">730×166 25.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>But when you connect your commands with &amp;&amp; , exit 1 will always be executed.<br>
Could you please change that line to the next syntax?</p>
<pre><code class="lang-auto">TAG_LIST=$(echo $RELEASE_VERSION | tr '.' ' ')
</code></pre>

**Comment by Jojoooo1** [comment, marked as answer]
<p>Hi <a class="mention" href="/yanjingzhu">@yanjingzhu</a>,</p>
<p>My response from days ago seems to not have been registered :(. Anyway basically my IF condition should have works since it was measuring the length of the TAG_LIST variable.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/a/a368e52febd6ae400de34c1a11afdd8a99a26fab.png" data-download-href="/uploads/short-url/njAz0SUWXd8RMpuzSUJO9j9KIFd.png?dl=1" title="Capture du 2020-05-29 14-32-44" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/17641873/181090761-07dbee2b-9ab2-4f3b-9c9c-8499675b3e5d.png" alt="Capture du 2020-05-29 14-32-44" data-base62-sha1="njAz0SUWXd8RMpuzSUJO9j9KIFd" width="690" height="77" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/a/a368e52febd6ae400de34c1a11afdd8a99a26fab_2_690x77.png, https://aws1.discourse-cdn.com/github/optimized/2X/a/a368e52febd6ae400de34c1a11afdd8a99a26fab_2_1035x115.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/a/a368e52febd6ae400de34c1a11afdd8a99a26fab.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/a/a368e52febd6ae400de34c1a11afdd8a99a26fab_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Capture du 2020-05-29 14-32-44</span><span class="informations">1336×150 51.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I was able to solve it using this which make sens from an array stand of point by using the two parentheses:</p>
<pre><code class="lang-auto">TAG_LIST=($(echo $NEW_RELEASE_VERSION | tr '.' ' '))
</code></pre>
<p>All the best,</p>
<p>Jonathan.</p>
